brigade.js

```diff
@@ -127,10 +128,169 @@ function ghNotify(state, msg, e, project) {
+      `docker build -t ${imgName} .`,
+      `docker tag ${imgName} ${latest}`,
+      `docker push ${imgName}`,
+      `echo '<======== Finished ${i}'`,
+      `cd ..`
+    );
+  }
+  dind.tasks.push("killall dockerd");
+
+  return Group.runEach([goBuild, dind]);
+}
+
+function acrBuild(e, project, tag) {
+  const gopath = "/go"
+  const localPath = gopath + "/src/github.com/" + project.repo.name;
+  const org = "deis";
+  const images = [
+    //"brig", // Uncomment this after 0.13.0.
+    "brigade-api",
+    "brigade-controller",
+    "brigade-cr-gateway",
+    "brigade-vacuum",
+    "brigade-worker", // brigade-worker does not have a rootfs. Could probably minify src into one and save space
+    "git-sidecar",
+    "brigade-github-gateway"
+  ]
+
+  // We build in a separate pod b/c AKS's Docker is too old to do multi-stage builds.
+  const goBuild = new Job("brigade-build", goImg);
+  goBuild.storage.enabled = true;
+  goBuild.env = {
+    "DEST_PATH": localPath,
+    "GOPATH": gopath
+  };
+  goBuild.tasks = [
+    `cd /src && git checkout ${tag}`,
+    "go get github.com/golang/dep/cmd/dep",
+    `mkdir -p ${localPath}/bin`,
+    `mv /src/* ${localPath}`,
+    `cd ${localPath}`,
+    "dep ensure",
+    "make build-docker-bins"
+  ];
+
+  for (let i of images) {
+    goBuild.tasks.push(
+      // Copy the Docker rootfs of each binary into shared storage. This is
+      // a little tricky because worker is non-Go, so later we will have
+      // to copy them back.
+      `mkdir -p /mnt/brigade/share/${i}/rootfs`,
+      // If there's no rootfs, we're done. Otherwise, copy it.
+      `[ ! -d ${i}/rootfs ] || cp -a ./${i}/rootfs/* /mnt/brigade/share/${i}/rootfs/`,
+    );
+  }
+  goBuild.tasks.push("ls -lah /mnt/brigade/share");
+
+  let registry = "brigade"
+  var builder = new Job("az-build", "microsoft/azure-cli:latest")
+  builder.storage.enabled = true;
+  builder.tasks = [
+    // When the extension is included by default, we can remove this.
+    "az extension add --source https://acrbuild.blob.core.windows.net/cli/acrbuildext-0.0.4-py2.py3-none-any.whl -y",
+    // Create a service principal and assign it proper perms on the ACR.
+    `az login --service-principal -u ${project.secrets.acrName} -p '${project.secrets.acrToken}' --tenant ${project.secrets.acrTenant}`,
+    `cd /src`,
+    `mkdir -p ./bin`
+  ]
+
+  // For each image we want to build, build it, then tag it latest, then post it to registry.
+  for (let i of images) {
+    let imgName = org +"/"+i+":"+tag;
+    let latest = org +"/"+i+":latest";
+    builder.tasks.push(
+      `cd ${i}`,
+      `echo '========> Building ${i}'`,
+      `cp -a /mnt/brigade/share/${i}/rootfs ./rootfs`,
+      // Total hack to work around something weird with brigade-github-gateway
+      `[ ! -f /mnt/brigade/share/${i}/rootfs/${i} ] || cp /mnt/brigade/share/${i}/rootfs/${i} ./rootfs/`,
+      // Currently, it appears this is the only way to add two tags.
```

Comment:
💄 You can remove this comment since it's not applicable anymore. I forgot to delete it from my previous PR.